extension/jaegerremotesampling/provider_test.go

```diff
@@ -0,0 +1,557 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package jaegerremotesampling
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/testutils"
```

Comment:
Not too sure what to do here
This is because this test requires jaeger's Logger. Should I also add the logger code into extention/jaegerremotesampling?

Reply:
zap library has its own test utils in `zaptest/observer`. Here's a sample snippet from Jaeger code:
```
	zapCore, logs := observer.New(zap.InfoLevel)
	logger := zap.New(zapCore)
...
	message := logs.FilterMessage("Admin server started")
	assert.Equal(t, 1, message.Len(), "Expected Admin server started log message.")

	onlyEntry := message.All()[0]
	hostPort := onlyEntry.ContextMap()["http.host-port"].(string)
	port, _ := strconv.Atoi(strings.Split(hostPort, ":")[3])
	assert.Positive(t, port)
```

Reply:
Thanks for this!